cmds/mkdir/mkdir.go

```diff
@@ -16,32 +16,77 @@ package main
 import (
 	"flag"
 	"fmt"
+	"log"
 	"os"
+	"strconv"
+)
+
+const (
+	cmd                 = "mkdir [-m mode] [-v] [-p] <directory> [more directories]"
+	DefaultCreationMode = 0777
+	StickyBit           = 01000
+	SgidBit             = 02000
+	SuidBit             = 04000
 )
 
 var (
+	mode    = flag.String("m", "", "Directory mode")
 	mkall   = flag.Bool("p", false, "Make all needed directories in the path")
-	mode    = flag.Int("m", 0666, "Directory mode")
 	verbose = flag.Bool("v", false, "Print each directory as it is made")
 	f       = os.Mkdir
 )
 
+func init() {
+	// Usage Definition
+	defUsage := flag.Usage
+	flag.Usage = func() {
+		os.Args[0] = cmd
+		defUsage()
+	}
+}
+
 func main() {
 	flag.Parse()
 	if len(flag.Args()) < 1 {
-		fmt.Printf("Usage: mkdir [-m mode] [-v] [-p] <directory> [more directories]\n")
+		flag.Usage()
 		os.Exit(1)
 	}
 	if *mkall {
 		f = os.MkdirAll
 	}
+
+	// Get Correct Creation Mode
+	var m uint64
+	var err error
+	if *mode == "" {
+		m = DefaultCreationMode
+	} else {
+		m, err = strconv.ParseUint(*mode, 8, 32)
+		if err != nil || m > 07777 {
+			log.Fatalf("invalid mode '%s'", *mode)
+		}
+	}
+	createMode := os.FileMode(m)
+	if m&StickyBit != 0 {
+		createMode |= os.ModeSticky
+	}
+	if m&SgidBit != 0 {
+		createMode |= os.ModeSetgid
+	}
+	if m&SuidBit != 0 {
+		createMode |= os.ModeSetuid
+	}
+
 	for _, name := range flag.Args() {
-		if err := f(name, os.FileMode(*mode)); err != nil {
-			fmt.Printf("%v: %v\n", name, err)
+		if err := f(name, createMode); err != nil {
+			log.Printf("%v: %v\n", name, err)
```

Comment:
we can do this on a future PR, but the Go common way to do this is a log.Printf and a continue, then 
avoid the else.

Reply:
Fixed